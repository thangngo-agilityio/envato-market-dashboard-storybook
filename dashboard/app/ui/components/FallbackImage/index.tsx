import { Image, ImageProps } from '@chakra-ui/react';
import { IMAGES } from '@/lib/constants';

interface FallbackImageProps extends ImageProps {
  fallbackSrc?: string;
  fallbackStrategy?: 'onError';
}

const FallbackImage = ({
  fallbackSrc = IMAGES.FALLBACK.url,
  fallbackStrategy = 'onError',
  ...props
}: FallbackImageProps) => (
  <Image
    {...props}
    fallbackStrategy={fallbackStrategy}
    fallbackSrc={fallbackSrc}
  />
);

export default FallbackImage;
