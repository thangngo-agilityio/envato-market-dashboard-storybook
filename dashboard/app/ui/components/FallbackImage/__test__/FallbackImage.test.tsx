import { render } from '@testing-library/react';
import FallbackImage from '..';

describe('FallbackImage component', () => {
  it('Should render correctly', () => {
    const { container } = render(<FallbackImage alt="Test Image" />);

    expect(container).toMatchSnapshot();
  });
});
