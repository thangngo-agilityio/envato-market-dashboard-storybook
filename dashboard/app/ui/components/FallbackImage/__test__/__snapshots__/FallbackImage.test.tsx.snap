// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FallbackImage component Should render correctly 1`] = `
<div>
  <img
    alt="Test Image"
    class="chakra-image css-0"
  />
</div>
`;
